include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -164,19 +188,24 @@ namespace internal
 
                   for (unsigned int j = 0; j < this->finite_elements[i]->size();
                        ++j)
-                    if ((*this->finite_elements[i])[j].reference_cell_type() !=
+                    if ((*this->finite_elements[i])[j].reference_cell_type() ==
                         ReferenceCell::get_hypercube(dim))
-                      quadrature.push_back(*quadrature1);
-                    else
                       quadrature.push_back(*quadrature2);
+                    else if ((*this->finite_elements[i])[j]
+                               .reference_cell_type() ==
+                             ReferenceCell::Type::Wedge)
+                      quadrature.push_back(*quadrature3);
+                    else
+                      quadrature.push_back(*quadrature1);
 
                   x_fe_values[i] =
                     std::make_shared<dealii::hp::FEValues<dim, spacedim>>(
                       this->mapping_collection,
                       *this->finite_elements[i],
                       quadrature,
-                      do_simplex ? (update_flags | update_quadrature_points) :
-                                   update_flags);
+                      (do_simplex || do_wedge) ?
+                        (update_flags | update_quadrature_points) :
+                        update_flags);
```

Comment:
Can you add a comment why you need `update_quadrature_points` but only for some of the element types?